src/lib/helpers/typeahead.tsx

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import { Option } from "react-bootstrap-typeahead/types/types";
+import { Menu, MenuItem } from "react-bootstrap-typeahead";
+import { RenderMenuProps } from "react-bootstrap-typeahead/types/components/Typeahead";
+import { LabelValueOption } from "../types/LabelValueOption";
+
+const convertTypeaheadOptionsToStringArray = (options: Option[]): string[] => {
+  const isStringArray = options.length > 0 && options.every((value) => typeof value === "string");
+
+  if (isStringArray) {
+    return options as string[];
+  }
+
+  return (options as Record<string, string>[]).map((option) => option.value);
+};
+
+const renderMenu = (results: LabelValueOption[], menuProps: RenderMenuProps): JSX.Element => {
+  const anonymousOptions: LabelValueOption[] = [];
```

Comment:
This code can be written way better.. can you try to use the new Set(groups) to get distinct groups and directly render them?

Reply:
done